examples/app/Embeddings.ipynb

```diff
@@ -20,36 +48,257 @@
+    }
+   },
+   "cell_type": "markdown",
+   "id": "7619577f",
+   "metadata": {},
+   "source": [
+    "You can change the type of the displayed weave panel by changing `Table` to `Plot` and it will intelligently change to display a plot of the data.\n",
+    "![image.png](attachment:image.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "356dc2a9",
+   "metadata": {},
+   "source": [
+    "## Embed and Cluster the data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4fd47c57",
+   "metadata": {},
+   "source": [
+    "Now we can pass a column from this data to another weave `Op` to embed it. This will turn each row in the column `name` into a vector."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "43cec2d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "embeddings = openai.openai_embed(data['name'], {\"model\": \"text-embedding-ada-002\"})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6715de4a",
+   "metadata": {},
+   "source": [
+    "Now, we'll project this to 2 dimensions for clustering, and then we'll cluster this projection using `hdbscan`."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "500e19b1",
+   "id": "5eff5bca",
    "metadata": {},
    "outputs": [],
    "source": [
-    "data"
+    "clusterable_projection = umap.umap_project(\n",
+    "    embeddings, {\n",
+    "        'n_neighbors': 30,\n",
+    "        'min_dist': 0,\n",
+    "        'n_components': 2,\n",
+    "    }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "583830d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "clusters = hdbscan.hdbscan_cluster(clusterable_projection, {\n",
+    "            'min_samples': 10,\n",
+    "            'min_cluster_size': 60\n",
```

Comment:
would be super cool to see how one would do hyperparameter exploration here! two projection plots side by side with a different setting for min_samples? I think we have space in this tutorial, and we could show off the "duplicate" functionality after everything is configured? also not sure if you've already tried this/it doesn't yield anything interesting—also recognize this is not the focus :)

Reply:
yeah, probably out of scope for this. Might be a bit too overwhelming